### PR TITLE
Add a checker that raise an error if there are not cut-files for the selected elements

### PR DESCRIPTION
**Pull request by jkpnen**
FIX: #212

### CODE REVIEW

File: widgets/detector_settings.py [hunk 1]
```diff
@@ -519,14 +522,45 @@ def __add_efficiency(self):
...
+                    found_eff_files.append(cut_element_str)
+        for file in found_eff_files:
+            if current_eff_file.stem not in file:  # Gives a warning
+                # if a cut file is not found
+                reply = QtWidgets.QMessageBox.warning(
+                    self, "Warning",
+                    f"All measurements do not have a "
+                    f"cut file for the element "
+                    f"{eff_file.stem}.\nDo you want to "
+                    f"continue?",
```
Comment (tpitkanen) [inline widgets/detector_settings.py:551]: The wording on this warning is a bit ambiguous. Does it mean
1. "None of the measurements have a cut file for the element {eff_file.stem}.\nDo you want to continue?" or
2. "At least one of the measurements is missing a cut file for the element {eff_file.stem}\nDo You want to continue?"?

**Review by tpitkanen — CHANGES_REQUESTED** [review]
I tested this with **H.eff** and could not add the efficiency file if the *H* element wasn't already in a measurement.

# Steps to reproduce

1. Open a request with a measurement (e.g. gradu-2021-11-27-copy)
2. Open a measurement
3. Make sure that there are no corresponding selections/cuts for the efficiency that you are about to add
4. Request Settings... > Detector > Efficiencies > + (add efficiency)

## Expected behavior

Potku prompts for confirmation.

(Potku used to allow adding the efficiency file without asking anything.)

## Actual behavior

Nothing happens. (`__check_if_selected_elements_have_cut_files` returns `None` because `eff_files_elements` is empty. The inner `QMessageBox.warning` is not reached.)

## Fix

- always return something (in general: if one code path returns a value, all the other ones must return something too)
- rethink the `for file in eff_files_elements:` check so that the message box is reached

**Comment by jkpnen** [comment]
Thanks. Silly mistake.

File: widgets/detector_settings.py [hunk 2]
```diff
@@ -563,24 +563,46 @@ def __check_if_selected_elements_have_cut_files(self, cuts_list: List,
...
-                if reply == QtWidgets.QMessageBox.Cancel:
-                    return True
-                elif reply == QtWidgets.QMessageBox.Ok:
-                    self.obj.add_efficiency_file(eff_file_path)
-                    current_eff_files.add(current_eff_file)
-                    return False
-            else:
-                self.obj.add_efficiency_file(eff_file_path)
-                current_eff_files.add(current_eff_file)
+        if len(eff_files_elements) is 0:
```
Comment (tpitkanen) [inline widgets/detector_settings.py:566]: Don't compare numbers by identity (`is`), use equality (`==`). Using `is` is roughly the same as doing `id(x) == id(y)`. CPython caches the first 100 or so number objects, so this works, but it's only an implementation detail. Example of bad things happening:

```python
>>> a = 100
>>> a is 100
<stdin>:1: SyntaxWarning: "is" with a literal. Did you mean "=="?
True
>>> b = 1000
>>> b is 1000
<stdin>:1: SyntaxWarning: "is" with a literal. Did you mean "=="?
False
>>> b == 1000
True
```

```suggestion
        if len(eff_files_elements) == 0:
```

**Review by tpitkanen — APPROVED** [review]
Looks good to me. I'll push that one fix and then merge this.

(Now we could add a unit test for this, but testing the GUI is kind of complicated so let's just skip that.)

**Comment by jkpnen** [comment]
> Looks good to me. I'll push that one fix and then merge this.
> 
> (Now we could add a unit test for this, but testing the GUI is kind of complicated so let's just skip that.)

Okay, thanks.

**Pull request merged**